### PR TITLE
#167727736: User should get his requests

### PR DESCRIPTION
**Pull request by Mcdavid95**
#### What does this PR do?
- Adds endpoint/feature to get a user's request

#### Description
- create seeds to seed request table

- add getAll services using sequelize findAndCountAll

- add controller to get user requests

- write pagination functions to calculate offset and limit

- write unit test for my implementation

- Document the API endpoint using swagger

#### How can this be manually tested
- Clone the repo
- Install dependencies, setup database and run `npm start:dev`
- Using postman or any http request tool, make a get request to `/api/v1/requests/user/:userId` to get user details
- You can optionally pass in  `page` and `perPage` to the request as query string as page number and number of items per page respectively to paginate the request
- Navigate to `/api/v1/docs` to test further with swagger ui 

#### Relevant Pivotal tracker stories
- [#167727736](https://www.pivotaltracker.com/story/show/167727736)

### CODE REVIEW

File: src/database/seeders/20190819153902-user.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ module.exports = {
         lastName: 'Williams',
         email: 'jammy@gmail.com',
         phoneNo: '2347032123304',
-        password: 'jammy11',
+        password: 'jammy11167',
```
Comment (tejiri4) [inline src/database/seeders/20190819153902-user.js:11]: why change the password?

**Review by tejiri4 — APPROVED** [review]
Write test to increase coverage and rebase with develop

File: src/database/seeders/20190819153902-user.js (hunk 1 above)
Comment (Mcdavid95) [reply src/database/seeders/20190819153902-user.js:11]: It was failing due to the password length

File: src/utils/pagination.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+
+/**
+ * @function calculateLimitAndOffset
+ * @param {number} page page number to get
+ * @param {number} perPage number of items per page/request
+ * @param {number} defaultLimit default items per page = 50
+ * @returns {object} returns object containing limit and offset
+ */
+export const calculateLimitAndOffset = (page, perPage, defaultLimit) => {
```
Comment (Mcdavid95) [inline src/utils/pagination.js]: set defaultLimit = 20

File: src/utils/pagination.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  const offset = (page ? Number(page) - 1 : 0) * (perPage ? Number(perPage) : defaultLimit);
```
Comment (Mcdavid95) [inline src/utils/pagination.js:10]: validate for less than 1 or 0

**Review by ladaposamuel — APPROVED** [review]
Looks good

**Review by jsbuddy — APPROVED** [review]
LGTM 👌

**Review by komsic — APPROVED** [review]
LGTM

**Pull request merged**